### PR TITLE
internal: Clean up runnable lsp extension

### PR DESCRIPTION
**Pull request by Veykril**
This feels like a natural addition to me, and also allows us to drop the expect-test hardcoding from the extension. Additionally, `cargoExtraArgs` is pointless, all the client will do is merge it with `cargoArgs` so the server can do that instead of delegating that to the client.

### CODE REVIEW

File: docs/dev/lsp-extensions.md [hunk 1]
```diff
@@ -376,12 +376,29 @@ rust-analyzer supports two `kind`s of runnables, `"cargo"` and `"shell"`. The `a
 
 ```typescript
 {
+    /**
+     * Environment variables to set before running the command.
+     */
+    environment?: Record<string, string>;
```
Comment (Veykril) [inline docs/dev/lsp-extensions.md:382]: Open to alternative names, should we be explicit with `environmentVariables`?

File: docs/dev/lsp-extensions.md [hunk 2]
```diff
@@ -376,12 +376,29 @@ rust-analyzer supports two `kind`s of runnables, `"cargo"` and `"shell"`. The `a
...
+    /**
+     * The working directory to run the command in.
+     */
+    cwd: string;
```
Comment (Veykril) [inline docs/dev/lsp-extensions.md:386]: I feel like this being mandatory gives a more unique experience across clients. We should not leave this decision to clients as that could result in runnables sometimes working sometimes not. (notable it is required by the shell version already)

**Review by davidbarsky — APPROVED** [review]
this looks good and reasonable to me.

File: docs/dev/lsp-extensions.md (hunk 1 above)
Comment (davidbarsky) [reply docs/dev/lsp-extensions.md:382]: could also call it `envVariables`?

File: docs/dev/lsp-extensions.md (hunk 2 above)
Comment (davidbarsky) [reply docs/dev/lsp-extensions.md:386]: yeah, reducing the number of possible options is great. the combinatorial explosion of options made this really difficult to reason about.

**Comment by ia0** [comment]
I'm a bit confused by https://github.com/rust-lang/rust-analyzer/issues/4604#issuecomment-2211788144, in particular:

> gets rid of `cargoExtraArgs` and bakes that into `cargoArgs`

Does this refer to `rust-analyzer.cargo.extraArgs`? I don't see any `rust-analyzer.cargo.args` in the [manual](https://rust-analyzer.github.io/manual.html). Or does this refer to something else?

**Comment by davidbarsky** [comment]
> I'm a bit confused by [#4604 (comment)](https://github.com/rust-lang/rust-analyzer/issues/4604#issuecomment-2211788144), in particular:
> 
> > gets rid of `cargoExtraArgs` and bakes that into `cargoArgs`
> 
> Does this refer to `rust-analyzer.cargo.extraArgs`? 

It does not. It refers to the field in the runnable sent from the server to the client.

**Comment by Veykril** [comment]
It refers to this https://github.com/rust-lang/rust-analyzer/blob/78d5f05e704dfe54ac70806973876608af3e2ba3/docs/dev/lsp-extensions.md#L382 which is the `rust-analyzer.cargo.extraArgs` config yes. `rust-analyzer.cargo.args` isn't a thing, as the `cargoArgs` in the runnable is populated by the server depending on the task, its not something the user configures.

File: docs/dev/lsp-extensions.md (hunk 1 above)
Comment (Veykril) [reply docs/dev/lsp-extensions.md:382]: Eh, I'll stick with `environment` I think.

Comment (davidbarsky) [reply docs/dev/lsp-extensions.md:382]: fair enough! not terribly attached.

**Comment by ia0** [comment]
Thanks a lot for the clarification!

**Pull request merged**